### PR TITLE
Add a method that unpacks vertices and indices into mesh.rs

### LINKED ISSUE #126: Add a method that unpacks vertices and indices into mesh.rs

### PR DESCRIPTION
**Pull request by ZKpot**
closes #126

### CODE REVIEW

File: dotrix_core/src/renderer.rs [hunk 1]
```diff
@@ -301,6 +301,22 @@ impl AttributeFormat {
             AttributeFormat::Uint32x4 => 4 * 4,
         }
     }
+
+    /// Returns the actual attribute TypeId
+    pub fn format(&self) -> std::any::TypeId {
```
Comment (Lowentwickler) [inline dotrix_core/src/renderer.rs:306]: call it `type_id` instead of `format` please

File: dotrix_core/src/assets/mesh.rs [hunk 2]
```diff
@@ -44,11 +44,38 @@ impl Mesh {
         self.layout.push(format);
     }
 
-    /// Sets indicies to the mesh
+    /// Get vertices with type casting
+    pub fn vertices_as<T>(&self, index: usize) -> Vec<T>
```
Comment (Lowentwickler) [inline dotrix_core/src/assets/mesh.rs:48]: both methods should return Iterators instead.

**Review by Lowentwickler — CHANGES_REQUESTED** [review]

**Comment by ZKpot** [comment]
@Lowentwickler any ideas why *Style / Rust-fmt (pull_request)*  is failing? Couldn't really figure that out from the details.

File: dotrix_core/src/assets/mesh.rs (hunk 2 above)
Comment (ZKpot) [reply dotrix_core/src/assets/mesh.rs:48]: should I simply remove .collect or you see it somehow more complicated?

**Comment by Lowentwickler** [comment]
> @Lowentwickler any ideas why _Style / Rust-fmt (pull_request)_ is failing? Couldn't really figure that out from the details.

just run `cargo fmt` in the repository root

File: dotrix_core/src/assets/mesh.rs (hunk 2 above)
Comment (Lowentwickler) [reply dotrix_core/src/assets/mesh.rs:48]: I would make a wrapper around the vector's mapped iterator

**Review by Lowentwickler — APPROVED** [review]

**Pull request merged**